### PR TITLE
Save all build artifacts that are copied to file share as pipeline artifact

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/818
Fixes: https://github.com/NuGet/Client.Engineering/issues/740

Regression? No

## Description

* Create a build variable with the value "RTM" or "NonRTM", so we don't have to duplicate every task, with conditions saying whether each should work on RTM or NonRTM builds
* publish `artifacts\vs15` as pipeline artifact on both rtm and non-rtm builds
* publish nupkgs as pipeline artifacts
* publish symbols as pipeline artifacts
* publish legacy localization files as pipeline artifacts

## PR Checklist

- [X] PR has a meaningful title
- [X] PR has a linked issue.
- [X] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [X] N/A (Infrastructure)

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [X] N/A

### CODE REVIEW

**Comment by zivkan** [comment]
@donnie-msft this should fix an issue currently assigned to you

**Comment by donnie-msft** [comment]
@zivkan assigned it to you

File: eng/pipelines/templates/Build_and_UnitTest.yml [hunk 1]
```diff
@@ -485,6 +498,12 @@ steps:
     TargetFolder: "$(BuildOutputTargetPath)\\symbols-rtm"
   condition: " and(succeeded(), not(eq(variables['BuildRTM'], 'false')), eq(variables['IsOfficialBuild'], 'true')) "
 
+- task: PublishPipelineArtifact@1
+  displayName: "Publish symbols as pipeline artifacts"
+  inputs:
+    targetPath: "$(Build.Repository.LocalPath)\\artifacts\\symbolstoindex"
+    artifactName: "symbols - $(RtmLabel)"
```
Comment (donnie-msft) [inline eng/pipelines/templates/Build_and_UnitTest.yml:513]: Should this match legacy `Copy Symbols` task only copy when Official Build is true?

File: eng/pipelines/templates/Build_and_UnitTest.yml [hunk 2]
```diff
@@ -309,6 +309,13 @@ steps:
     TargetFolder: "$(BuildOutputTargetPath)\\artifacts\\$(VsixPublishDir)\\$(NupkgOutputDir)"
   condition: and(succeeded(), or(eq(variables['IsOfficialBuild'], 'true'), eq(variables['BuildRTM'], 'true')))  #skip this task for nonRTM in private build
 
+- task: PublishPipelineArtifact@1
+  displayName: "Publish nupkgs"
+  inputs: 
+    targetPath: "$(Build.Repository.LocalPath)\\artifacts\\$(NupkgOutputDir)"
```
Comment (donnie-msft) [inline eng/pipelines/templates/Build_and_UnitTest.yml:315]: There's also a step `displayName: "Copy NuGet.exe, VSIX and EndToEnd"`
which copies from the VSIX, nuget.exe, etc from a higher folder. Will those be captured as artifacts somewhere?

File: eng/pipelines/templates/Build_and_UnitTest.yml [hunk 3]
```diff
@@ -449,6 +455,13 @@ steps:
     artifactName: "localizeInputs"
   condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
 
+- task: PublishPipelineArtifact@1
+  displayName: "Upload legacy localizeInputs artifact"
+  inputs:
+    targetPath: "$(Build.Repository.LocalPath)\\artifacts\\**\\*.lcg;!$(Build.Repository.LocalPath)\\artifacts\\localize\\**\\*"
+    artifactName: "legacy localizeInputs"
+  condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'))"
```
Comment (donnie-msft) [inline eng/pipelines/templates/Build_and_UnitTest.yml]: Official only?

File: eng/pipelines/templates/Build_and_UnitTest.yml (hunk 2 above)
Comment (donnie-msft) [reply eng/pipelines/templates/Build_and_UnitTest.yml:315]: > publish artifacts\vs15 as pipeline artifact on both rtm and non-rtm builds

where's this?

I thought artifacts\VS15\Nupkgs
was `\\artifacts\\$(VsixPublishDir)\\$(NupkgOutputDir)`
meaning you're publishing `\\artifacts\\Nupkgs` ?

Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:315]: Lines 427 to 433 define the "publish artifacts\VS15 directory" task. I deleted the condition that only does it on the non-RTM build, so we get RTM artifacts as well.

File: eng/pipelines/templates/Build_and_UnitTest.yml (hunk 3 above)
Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml]: I intended to leave out official for testing, and once validated, put it in before merging. I'll add a comment in my next push to make it obvious, and hopefully remember to fix before merging.

File: eng/pipelines/templates/Build_and_UnitTest.yml (hunk 2 above)
Comment (donnie-msft) [reply eng/pipelines/templates/Build_and_UnitTest.yml:315]: Line 427 shows as `'OptProfV2:  publish profiling inputs to artifact services'` for me, atm.
Anyway, I do see the VSIX/exe in your run's artifacts.
I do not see published nupkgs, though 😮

Comment (donnie-msft) [reply eng/pipelines/templates/Build_and_UnitTest.yml:315]: Nevermind - I see that step didn't happen due to failed tests. 👀

Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:315]: Sorry, line 427-433 in the "previous" version, it's likes 434-439 in the new file.

Comment (donnie-msft) [reply eng/pipelines/templates/Build_and_UnitTest.yml:315]: Ok, now your build shows the nupkgs for RTM. Did you intend that we no longer keep Non-RTM nupkgs?

Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:315]: On private builds we don't generate nupkgs in the non-rtm build (finish as quickly as possible, so mac, apex, and e2e tests can start)

**Review by donnie-msft — APPROVED** [review]

**Review by dominoFire — APPROVED** [review]

**Pull request merged**